### PR TITLE
Improve visibility of Redis counter

### PR DESCRIPTION
**Pull request by klaidliadon**
Adding log to httprate redis plus some cleanup.

- moved some types to internal package (duration, cycle store, usage tracker)
- moved memory store to test package
- replace error handler with option in middleware (like authcontrol)
- use option logger in httprate redis onError and onFallback.

### CODE REVIEW

File: middleware/middleware_permission.go [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+	eh := errHandler
+	if o != nil && o.ErrHandler != nil {
+		eh = o.ErrHandler
 	}
 
+	logger := logger.NewLogger(logger.LogLevel_DEBUG)
+	if o != nil && o.Logger != nil {
+		logger = o.Logger
+	}
+	logger = logger.With(slog.String("middleware", "ratelimit"))
```
Comment (david-littlefarmer) [inline middleware/middleware_permission.go:22]:
```suggestion
	logger = logger.With(slog.String("middleware", "ensurePermission"))
```
Here is already same static log property:value https://github.com/0xsequence/quotacontrol/pull/57/files#diff-89099739e0265112baf868728d4eb7b062297a5e722c7d55e43f51626987ca66R69

**Review by david-littlefarmer — COMMENTED** [review]
1st round of CR

File: internal/usage/usage.go [hunk 2]
```diff
@@ -15,33 +15,39 @@ type UsageUpdater interface {
 	UpdateProjectUsage(ctx context.Context, service proto.Service, now time.Time, usage map[uint64]*proto.AccessUsage) (map[uint64]bool, error)
 }
 
-func newUsageRecord() usageRecord {
-	return usageRecord{
+func NewRecord() Record {
```
[the remark this answers is not in the document]
Comment (klaidliadon) [reply internal/usage/usage.go:18]: It's used in test memory store:
https://github.com/0xsequence/quotacontrol/blob/f6a3710bf2d9370f7c256f85abba49319a5e7e3b/test/mem.go#L18

File: internal/usage/usage.go [hunk 3]
```diff
@@ -15,33 +15,39 @@ type UsageUpdater interface {
...
 	}
 }
 
-type usageRecord struct {
+type Record struct {
 	ByProjectID map[uint64]*proto.AccessUsage
 	ByAccessKey map[string]*proto.AccessUsage
 }
 
+func NewTracker() *Tracker {
```
[the remark this answers is not in the document]
Comment (klaidliadon) [reply internal/usage/usage.go:30]: It's used here: https://github.com/0xsequence/quotacontrol/blob/f6a3710bf2d9370f7c256f85abba49319a5e7e3b/client.go#L56

File: internal/usage/usage.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (klaidliadon) [reply internal/usage/usage.go:18]: The package is internal anyway.

File: internal/usage/usage.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (klaidliadon) [reply internal/usage/usage.go:30]: The package is internal anyway.

File: config.go [hunk 4]
```diff
@@ -3,37 +3,39 @@ package quotacontrol
...
-	ErrorConfig   ErrorConfig         `toml:"error_config"`
+	Enabled       bool            `toml:"enabled"`
+	URL           string          `toml:"url"`
+	AuthToken     string          `toml:"auth_token"`
+	UpdateFreq    toml.Duration   `toml:"update_freq"`
+	RateLimiter   RateLimitConfig `toml:"rate_limiter"`
+	Redis         RedisConfig     `toml:"redis"`
+	DefaultUsage  *int64          `toml:"default_usage"`
+	LRUSize       int             `toml:"lru_size"`
+	LRUExpiration toml.Duration   `toml:"lru_expiration"`
```
Comment (david-littlefarmer) [inline config.go]: We don't need the `internal/toml/duration` this should work.
I tried it, and tests passed.
```suggestion
	UpdateFreq    time.Duration   `toml:"update_freq"`
	RateLimiter   RateLimitConfig `toml:"rate_limiter"`
	Redis         RedisConfig     `toml:"redis"`
	DefaultUsage  *int64          `toml:"default_usage"`
	LRUSize       int             `toml:"lru_size"`
	LRUExpiration time.Duration   `toml:"lru_expiration"`
```

**Review by david-littlefarmer — COMMENTED** [review]
last comment

**Review by david-littlefarmer — APPROVED** [review]
LGTM

**Pull request merged**